src/main/scala/pipedsl/codegen/bsv/BluespecGeneration.scala

```diff
@@ -619,42 +607,62 @@ object BluespecGeneration {
      * @param stg - The stage to process
      * @return Some(BSV) rule if the stage can be killed, else None
      */
-    private def getStageKillRule(stg: PStage): Option[BRuleDef] = {
-      val killConds = getKillConds(stg.getCmds)
+    private def getStageKillRule(stg: PStage, is_spec: Boolean, is_excepting: Boolean): List[BRuleDef] = {
```

Comment:
rename to "getStageKillRules" (with an 's') to reflect that it can return multiple rules.